test/index.test.ts

```diff
@@ -225,6 +225,103 @@ describe('snyk-config', () => {
 
     expect(config.foo).toEqual('bar');
   });
+
+  it('doest not parse array values from env vars', () => {
+    const testArr = [1, 2, 3];
+    const testArrStr = JSON.stringify(testArr);
+
+    process.env.SNYK_TEST_ARR = testArrStr;
```

Comment:
looks like a repeating pattern, might be nothing might also be an indicator of future test flakiness / oddness

Reply:
I believe this is taken care of in this `afterEach` callback function:
https://github.com/snyk/config/tree/main/test/index.test.ts#L10 

However, it only seems to cover the `SNYK_*` environment variables, without cleaning the `CONFIG_*` environment variables. I'll add a fix for that, thanks!

Reply:
> However, it only seems to cover the SNYK_* environment variables, without cleaning the CONFIG_* environment variables. I'll add a fix for that, thanks!

Introduced in ed633d6

Reply:
What's the rationale for excluding parsing array values @ofekatr? We were chatting in the team that we might need something like that when we need to put a rotatable secret value in our secret manager, i.e. a key -> multiple-string-values. 

Perhaps this isn't something the config library should be concerned with, but it would be nice if we had a mechanism for receiving multiple values for a single key so we have parity with our json config loading.

I'm not sure I see the use case for extending this to object parsing, but arrays perhaps should get special treatment, WDYT?

Reply:
The package will support parsing array values from environment variable configuration values and does not exclude these. When the parameter flag is enabled, any value provided via environment variable configurations will be attempted to be JSON-parsed.

Reply:
ah nice, I must be reading the tests wrong